qiskit/chemistry/components/variational_forms/chc.py

```diff
@@ -0,0 +1,194 @@
+    the UCC Ansatz for a lower CNOT count.
+
+    Note:
+        It is not particle number conserving and the accuracy of the approximation decreases
+        with the number of excitations.
+    """
+
+    def __init__(self, num_qubits: int, depth: int = 1, ladder: bool = False,
+                 excitations: Optional[List[List[int]]] = None,
+                 entangler_map: Optional[List[int]] = None,
+                 entanglement: str = 'full',
+                 initial_state: Optional[InitialState] = None) -> None:
+        """
+
+        Args:
+            num_qubits:
+            depth:
+            ladder:
+            excitations:
+            entangler_map:
+            entanglement:
+            initial_state:
+        """
+
+        super().__init__()
+        self._num_qubits = num_qubits
+        self._depth = depth
+        self._excitations = None
+        self._entangler_map = None
+        self._initial_state = None
+        self._ladder = ladder
+        self._num_parameters = len(excitations)*depth
+        self._excitations = excitations
+        self._bounds = [(-np.pi, np.pi)] * self._num_parameters
+        self._num_qubits = num_qubits
+        if entangler_map is None:
+            self._entangler_map = VariationalForm.get_entangler_map(entanglement, num_qubits)
+        else:
+            self._entangler_map = VariationalForm.validate_entangler_map(entangler_map, num_qubits)
+        self._initial_state = initial_state
+        self._support_parameterized_circuit = True
+
+    def construct_circuit(self, parameters: Union[np.ndarray, List[Parameter], ParameterVector],
+                          q: Optional[QuantumRegister] = None) -> QuantumCircuit:
+        """
+        Construct the variational form, given its parameters.
+
+        Args:
+            parameters: circuit parameters
+            q: Quantum Register for the circuit.
+
+        Returns:
+            QuantumCircuit: a quantum circuit with given `parameters`
+
+        Raises:
+            ValueError: the number of parameters is incorrect.
+            ValueError: only supports single and double excitations at the moment.
+        """
+
+        if len(parameters) != self._num_parameters:
+            raise ValueError('The number of parameters has to be {}'.format(self._num_parameters))
+
+        if q is None:
+            q = QuantumRegister(self._num_qubits, name='q')
+        if self._initial_state is not None:
+            circuit = self._initial_state.construct_circuit('circuit', q)
+        else:
+            circuit = QuantumCircuit(q)
+
+        count = 0
+        for _ in range(self._depth):
+            for idx in self._excitations:
+
+                if len(idx) == 2:
+
+                    i = idx[0]
+                    r = idx[1]
+
+                    circuit.u1(-parameters[count] / 4 + np.pi / 4, q[i])
+                    circuit.u1(-parameters[count] / 4 - np.pi / 4, q[r])
```

Comment:
The U1 gate is being deprecated you should use `p` (for phase-gate) instead, see also Qiskit/qiskit-terra#5083.
```suggestion
                    circuit.p(-parameters[count] / 4 + np.pi / 4, q[i])
                    circuit.p(-parameters[count] / 4 - np.pi / 4, q[r])
```